mail-api.js

```diff
@@ -96,6 +96,34 @@ module.exports = {
     });
   },
   
+
+  getMessageAttachmentId: function(parameters, callback) {
+    var userSpec = utilities.getUserSegment(parameters);
+    var requestUrl = base.apiEndpoint() + userSpec + '/messages/' + parameters.messageId + '/attachments';
+    var apiOptions = {
+      url: requestUrl,
+      token: parameters.token,
+      user: parameters.user
+    };
+
+    base.makeApiCall(apiOptions, function(error, response) {
```

Comment:
You should support optional query parameters so callers can do things like `$select`. You can follow the pattern in `getMessages` for example:

```
if (parameters.odataParams !== undefined) {
    apiOptions['query'] = parameters.odataParams;
}
```